velox/dwio/parquet/common/BitStreamUtilsInternal.h

```diff
@@ -206,281 +208,276 @@ class BitReader {
-    buffered_values_ = ::arrow::bit_util::ToLittleEndian(buffered_values_);
-    memcpy(buffer_ + byte_offset_, &buffered_values_, 8);
-    buffered_values_ = 0;
-    byte_offset_ += 8;
-    bit_offset_ -= 64;
-    buffered_values_ =
-        (num_bits - bit_offset_ == 64) ? 0 : (v >> (num_bits - bit_offset_));
+  bufferedValues_ |= v << bitOffset_;
+  bitOffset_ += numBits;
+
+  if (FOLLY_UNLIKELY(bitOffset_ >= 64)) {
+    // Flush bufferedValues_ and write out bits of v that did not fit
+    bufferedValues_ = ::arrow::bit_util::ToLittleEndian(bufferedValues_);
+    memcpy(buffer_ + byteOffset_, &bufferedValues_, 8);
+    bufferedValues_ = 0;
+    byteOffset_ += 8;
+    bitOffset_ -= 64;
+    bufferedValues_ =
+        (numBits - bitOffset_ == 64) ? 0 : (v >> (numBits - bitOffset_));
   }
-  DCHECK_LT(bit_offset_, 64);
+  VELOX_DCHECK_LT(bitOffset_, 64);
   return true;
 }
 
 inline void BitWriter::Flush(bool align) {
-  int num_bytes =
-      static_cast<int>(::arrow::bit_util::BytesForBits(bit_offset_));
-  DCHECK_LE(byte_offset_ + num_bytes, max_bytes_);
-  auto buffered_values = ::arrow::bit_util::ToLittleEndian(buffered_values_);
-  memcpy(buffer_ + byte_offset_, &buffered_values, num_bytes);
+  int numBytes = static_cast<int>(::arrow::bit_util::BytesForBits(bitOffset_));
+  VELOX_DCHECK_LE(byteOffset_ + numBytes, maxBytes_);
+  auto bufferedValues = ::arrow::bit_util::ToLittleEndian(bufferedValues_);
+  memcpy(buffer_ + byteOffset_, &bufferedValues, numBytes);
 
   if (align) {
-    buffered_values_ = 0;
-    byte_offset_ += num_bytes;
-    bit_offset_ = 0;
+    bufferedValues_ = 0;
+    byteOffset_ += numBytes;
+    bitOffset_ = 0;
   }
 }
 
-inline uint8_t* BitWriter::GetNextBytePtr(int num_bytes) {
+inline uint8_t* BitWriter::GetNextBytePtr(int numBytes) {
   Flush(/* align */ true);
-  DCHECK_LE(byte_offset_, max_bytes_);
-  if (byte_offset_ + num_bytes > max_bytes_)
+  VELOX_DCHECK_LE(byteOffset_, maxBytes_);
+  if (byteOffset_ + numBytes > maxBytes_)
     return NULL;
-  uint8_t* ptr = buffer_ + byte_offset_;
-  byte_offset_ += num_bytes;
+  uint8_t* ptr = buffer_ + byteOffset_;
+  byteOffset_ += numBytes;
   return ptr;
 }
 
 template <typename T>
-inline bool BitWriter::PutAligned(T val, int num_bytes) {
-  uint8_t* ptr = GetNextBytePtr(num_bytes);
+inline bool BitWriter::PutAligned(T val, int numBytes) {
+  uint8_t* ptr = GetNextBytePtr(numBytes);
   if (ptr == NULL)
     return false;
   val = ::arrow::bit_util::ToLittleEndian(val);
-  memcpy(ptr, &val, num_bytes);
+  memcpy(ptr, &val, numBytes);
   return true;
 }
 
 namespace detail {
 
 template <typename T>
 inline void GetValue_(
-    int num_bits,
+    int numBits,
     T* v,
-    int max_bytes,
+    int maxBytes,
     const uint8_t* buffer,
-    int* bit_offset,
-    int* byte_offset,
-    uint64_t* buffered_values) {
+    int* bitOffset,
+    int* byteOffset,
+    uint64_t* bufferedValues) {
 #ifdef _MSC_VER
```

Comment:
nit: can remove the block under _MSC_VER since Velox does not support the Microsoft compiler.